### PR TITLE
Isolate declarativebookmarkmodel test cases

### PR DESCRIPTION
**Pull request by rojkov**

### CODE REVIEW

File: tests/auto/tst_declarativebookmarkmodel/tst_declarativebookmarkmodel.cpp [hunk 1]
```diff
@@ -9,211 +9,227 @@
...
+    QString m_bookmarksFile;
 };
 
-
-tst_declarativebookmarkmodel::tst_declarativebookmarkmodel()
-    : TestObject(QML_SNIPPET)
+void tst_declarativebookmarkmodel::initTestCase()
 {
-    bookmarkModel = TestObject::qmlObject<DeclarativeBookmarkModel>("bookmarkModel");
+    QString settingsLocation = QStandardPaths::writableLocation(QStandardPaths::DataLocation);
```
Comment (rainemak) [inline tests/auto/tst_declarativebookmarkmodel/tst_declarativebookmarkmodel.cpp]: Maybe worth adding in another PR something like ```BrowserPaths``` class that would have static methods like.

```
static QString downloadLocation()
static QString picturesLocation()
static QString dataLocation()
...
```
This way we could clean up a bit the webutils thingie and unit test cases would not contain assumptions about paths.

Comment (rojkov) [inline tests/auto/tst_declarativebookmarkmodel/tst_declarativebookmarkmodel.cpp]: Yep, I agree.

File: tests/auto/tst_declarativebookmarkmodel/tst_declarativebookmarkmodel.pro [hunk 2]
```diff
@@ -1,15 +1,13 @@
 TARGET = tst_declarativebookmarkmodel
 
-QT += quick concurrent
+QT += network concurrent
```
Comment (rainemak) [inline tests/auto/tst_declarativebookmarkmodel/tst_declarativebookmarkmodel.pro]: What component requires this one?

**Comment by rainemak** [comment]
Added a small question? Works as specified, LGTM. If question don't trigger actions, go ahead and merge.

File: tests/auto/tst_declarativebookmarkmodel/tst_declarativebookmarkmodel.pro (hunk 2 above)
Comment (rojkov) [inline tests/auto/tst_declarativebookmarkmodel/tst_declarativebookmarkmodel.pro]: `src/bookmarks/iconfetcher.cpp` (included by `src/bookmarks/bookmarks.pri`) needs it. I think it makes sense to refactor the tests not to depend on iconfetcher, but probably in a separate PR.

Comment (rainemak) [inline tests/auto/tst_declarativebookmarkmodel/tst_declarativebookmarkmodel.pro]: Yeap. Let's have a separate PR about that. So looks good.

**Pull request merged**